### PR TITLE
Reland NestedTensor Softmax (#83435)

### PR DESCRIPTION
**Pull request by huydhn**
Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/83435

Simple mask compute and softmax

Test Plan: unit test

Differential Revision: D38711915

fbshipit-source-id: 1e1ef67b23e93f567763b33588fb88b3edf5020c

Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by huydhn** [comment]
I realize that I don't need this additional PR and can just reland the original one https://github.com/pytorch/pytorch/pull/83435
